nixos/modules/image/repart-image.nix

```
@@ -78,53 +76,87 @@ let
     "xz" = "xz --keep --verbose --threads=0 -${toString compression.level}";
   }."${compression.algorithm}";
 in
-
-runCommand imageFileBasename
-{
+  stdenvNoCC.mkDerivation (finalAttrs:
+  (if (version != null)
+  then { pname = name; inherit version; }
```

This is rather confusing, we should use pname+version

```suggestion
  then { inherit pname version; }
```

---

This was already discussed in previous comments